polynote-spark/src/main/scala/polynote/kernel/lang/python/PySparkInterpreter.scala

```diff
@@ -0,0 +1,83 @@
+package polynote.kernel.lang.python
+
+import java.io.File
+
+import cats.effect.IO
+import cats.syntax.apply._
+import org.apache.spark.api.java.JavaSparkContext
+import org.apache.spark.sql.SparkSession
+import polynote.kernel.lang.LanguageInterpreter
+import polynote.kernel.util.KernelContext
+import py4j.GatewayServer
+
+class PySparkInterpreter(ctx: KernelContext) extends PythonInterpreter(ctx) {
+
+  override def sharedModules: List[String] = "pyspark" :: super.sharedModules
+
+  override def init(): IO[Unit] = super.init() *> withJep {
+    try {
+
+      // initialize py4j and pyspark in the way they expect
```

Comment:
Yeesh, that's a lot of stuff that we need to do! Do you have a link to docs or wherever you found this stuff?